### DISCUSSION TITLE
win10/ide-startup工程，git终端创建失败askpass.sh: No such file or directory，不能执行git clone命令

### DISCUSSION THREAD

**Discussion by CiciDongxue** [opensumi/core#2337] [Q&A] [answered]
win10，必须装visual studio，本地启动ide-startup工程后调出终端。

然后页面编辑器中新建终端，自动调出powershell.exe，输入“git clone xxx.git”，提示：
Cloning into 'xxx'...
error: cannot spawn xxxx\ide-startup\extensions\vscode.git\dist\askpass.sh: No such file or directory
fatal: could not read Username for 'http://gitlab.xxxx: No such file or directory

在编辑器终端回到c盘的用户目录，执行“git clone xxx.git”，提示：
Cloning into 'xxx'...
error: cannot spawn c:\ProgramFiles\VSCode\Microsoft VS Code\resources\app\extensions\git\dist\askpass.sh: No such file or directory
fatal: could not read Username for 'http://gitlab.xxx': No such file or directory

请问大家有遇到吗？是怎么解决的？

**Comment by erha19** [comment]
@CiciDongxue 有截图信息吗？同时使用的 vscode.git 插件版本是？

**Reply by CiciDongxue** [reply]
在sumi编辑器vscode.git中：git插件版本 v1.62.3，gitui插件版本是1.55.2

我自己本地电脑安装的git version 2.31.1.windows.1

**Reply by erha19** [reply]
看这里报错是插件文件路径找不到导致的报错问题，目前手头没机器验证 Windows 下的情况，如果使用的 2.21+ OpenSumi 版本，你可以安装一下高版本的 Git 插件尝试一下

https://open-vsx.org/extension/vscode/git
https://open-vsx.org/extension/vscode/git-base

**Reply by CiciDongxue** [reply]
sumi是2.21.8，安装插件git升级也无效。

用win10的管理员账号启动项目，仍然找不到路径，见下图。
浏览器报错截图：
![image](https://user-images.githubusercontent.com/30331842/221772633-64540484-67fe-4ed9-9453-8076ffb08ac4.png)
cmd报错截图（powershell同）：
![image](https://user-images.githubusercontent.com/30331842/221772256-c077bb34-ee3f-4290-a82a-e51845a069c2.png)

此路径确实存在，见下图：
![image](https://user-images.githubusercontent.com/30331842/221772393-4b3b6aa0-f656-486b-ad5f-2cba1b733999.png)

同时不能打开git-bash终端，见下图：
![image](https://user-images.githubusercontent.com/30331842/221772930-f3a31258-725f-4789-a30c-33aea164a7e3.png)

**Reply by erha19** [reply]
@CiciDongxue 你脱离 IDE ，在本机 PowerShell 下能正常 Clone 项目吗？感觉像是你本地 Git 被绑定了一些执行脚本

**Reply by CiciDongxue** [reply]
在单独的powershell中是可以clone成功的（见下图）。在sumi编辑器中，起的终端不可以，执行git clone命令不能加载git输入用户名和密码模块，git-bash终端也起不来。我跟我同事的win10都一样问题。
补充，mac电脑，启动项目、clone项目都没任何问题。

![image](https://user-images.githubusercontent.com/30331842/221803279-c8d7f0e6-3c45-4aba-ab09-a0cf629c097a.png)

**Comment by Ricbet** [comment, marked as answer]
看一下你下载后的 git 插件有没有这个文件 dist\askpass.sh

**Reply by CiciDongxue** [reply]
在ide-startup\extensions\vscode.git\dist目录中有askpass.sh文件

**Reply by shilin8805** [reply]
windows上需要给终端加个环境变量，参考[Demo](https://github.com/opensumi/opensumi-module-samples/blob/main/modules/terminal-env/browser/terminal-env.contribution.ts)，补充下面的代码试一下
```js
    this.injector.overrideProviders({
      token: ITerminalClientFactory2,
      useFactory: (injector) => async (widget: IWidget, options?: ICreateTerminalOptions) => {
        const terminalOptions: ICreateTerminalOptions = options || {}
        terminalOptions.id = widget.id
        const defaultProfile = await this.terminalProfileInternalService.resolveDefaultProfile()
        terminalOptions.config = defaultProfile
        if (process.platform === 'win32') {
          const defaultTerminalEnvironmentVariables = {} as any
          const gitInfo = await findGit()
          const gitPath = path.join(gitInfo.path, '..', '..', 'bin')
          const envPath = process.env.PATH || process.env.Path
          defaultTerminalEnvironmentVariables.PATH = `${envPath};${gitPath}`
          if (terminalOptions.config) {
            terminalOptions.config.env = {
              ...terminalOptions.config?.env,
              ...defaultTerminalEnvironmentVariables
            }
          }
        }
        return TerminalClientFactory.createClient2(injector, widget, terminalOptions)
      }
    })
```
